src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -2230,4 +2289,40 @@ static long mismatch(MemorySegment srcSegment, long srcFromOffset, long srcToOff
                 dstSegment, dstFromOffset, dstToOffset);
     }
 
+    /**
+     * A scope models the <em>lifetime</em> of all the memory segments associated with it. That is, a memory segment
+     * cannot be accessed if its associated scope is not {@linkplain #isAlive() alive}. A new scope is typically
+     * obtained indirectly, by creating a new {@linkplain Arena arena}.
+     * <p>
+     * Scope instances can be compared for equality. That is, two scopes
+     * are considered {@linkplain #equals(Object)} if they denote the same lifetime.
+     */
+    @PreviewFeature(feature=PreviewFeature.Feature.FOREIGN)
+    sealed interface Scope permits MemorySessionImpl {
+        /**
+         * {@return {@code true}, if the regions of memory backing the memory segments associated with this scope are
+         * still valid}
+         */
+        boolean isAlive();
+
+        /**
+         * Returns {@code true}, if the provided object is also a scope, which models the same lifetime as that
+         * modelled by this scope. In that case, it is always the case that
+         * {@code this.isAlive() == ((Scope)that).isAlive()}.
+         * @param that the object to be tested.
+         * @return {@code true}, if the provided object is also a scope, which models the same lifetime as that
+         * modelled by this scope.
+         */
```

Comment:
A `@return` tag could be used here
```suggestion
        /**
         * {@return {@code true}, if the provided object is also a scope, which models the same lifetime as that
         * modelled by this scope.} In that case, it is always the case that
         * {@code this.isAlive() == ((Scope)that).isAlive()}.
         * @param that the object to be tested.
         */
```

Reply:
Oh - I didn't know you could use return tag plus other text...